src/pipecat/transports/services/livekit.py

```diff
@@ -0,0 +1,556 @@
+import asyncio
+from dataclasses import dataclass
+from typing import Any, Awaitable, Callable, List
+
+import numpy as np
+from loguru import logger
+from pipecat.frames.frames import (
+    AudioRawFrame,
+    CancelFrame,
+    EndFrame,
+    Frame,
+    MetricsFrame,
+    StartFrame,
+    TransportMessageFrame,
+)
+from pipecat.processors.frame_processor import FrameDirection, FrameProcessor
+from pipecat.transports.base_input import BaseInputTransport
+from pipecat.transports.base_output import BaseOutputTransport
+from pipecat.transports.base_transport import BaseTransport, TransportParams
+from pipecat.vad.vad_analyzer import VADAnalyzer
+from pydantic import BaseModel
+from scipy import signal
+from tenacity import retry, stop_after_attempt, wait_exponential
+
+try:
+    from livekit import rtc
+except ModuleNotFoundError as e:
+    logger.error(f"Exception: {e}")
+    logger.error("In order to use LiveKit, you need to `pip install pipecat-ai[livekit]`.")
```

Comment:
We probably want to add `livekit` to pyproject.toml with the right dependencies.